src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -106,72 +266,186 @@ public void run() {
                     return;
                 }
 
-                int level;
                 try {
                     if (command instanceof HSBType) {
-                        HSBType hsb = (HSBType) command;
-                        clusterColorControl
-                                .moveToHueAndSaturationCommand((int) (hsb.getHue().doubleValue() * 254.0 / 360.0 + 0.5),
-                                        (int) (hsb.getSaturation().doubleValue() * 254.0 / 100.0 + 0.5), 10)
-                                .get();
-                        level = hsb.getBrightness().intValue();
+                        HSBType color = (HSBType) command;
+                        PercentType brightness = color.getBrightness();
+
+                        if (brightness.intValue() != currentHSB.getBrightness().intValue()) {
+                            changeBrightness(brightness);
+                            // Wait for transition to complete
+                            // Some lights do not like receiving a level/color change command
+                            // while the previous transition is in progress...
+                            Thread.sleep(1000);
```

Comment:
I'm not sure that this will work properly under all conditions? This will sleep this thread, but I don't think there's anything in the binding will stop another command coming in in another thread and sending another brightness command.

I think a better implementation would be to record the last brightness time, and then check that before sending an update. I think in the near future we will want to make this configurable so that lamps that do allow it, can support it (don't worry about that right now though as I'm working on configuration at the moment).

Reply:
So do we leave it as is by now, or do you prefer to synchronize this part of the code ?

Reply:
I actually misunderstood the purpose of this...

Do you have bulbs that don't like color setting when brightness is being changed? What happens if you send these close together? What happens if you just send two brightness commands close together - is that ok or does it also cause problems?

Does the same issue occur if brightness is moved after color?

I don't really like the sleep here as I'm sure it must look a bit strange on some bulbs and if someone is moving a color wheel, then for sure we will violate this 1 second anyway and the problem will happen unless we delay everything which is (IMHO) not a nice option as we effectively slow down changes from a UI is someone is "spinning" a color wheel.

Reply:
My IKEA lights abort the brightness transition at whichever point it is when a color change is requested. The same if the order is reverted, with the color. Will experiment a bit more to see if I can workaround this.

Anyway, this delay should only happen when both color and brightness are changed within the same command (HSB Type):

    if (brightness.intValue() != currentHSB.getBrightness().intValue()) {

So when changing through a color wheel this branch is skipped as we only change color, not brightness, and we compare it to current value and skip the change (and delay) if brightness does not change

The 1 second delay will only happen then if we receive a command which changes color and brightness at the same time (which I do not think is so common, in terms of UI: might happen in rules), so it should have no impact on the normal behaviour of changes through UI (color wheel, etc)

I do not like the sleep either, and it could be made configurable when config is available, only for lights requiring it.

Reply:
What happens if you use a faster transition time? eg set transition time to 0 in this case and avoid the sleep? Would that work?

Reply:
I does not work either. It looks like the IKEA lights do not like receiveing both commands just one  immediately after the other. A small delay (tried with 100ms) is still necessary. When I was looking for DH examples for SmartThings, I say a similar issue (and workaround) on other device handlers, so I think this kind of bug may not be exclusive of IKEA lights, anyway...

Immediate transition time is, anyway, somehow "awful to see", used to having the lights change smoothly.

So what I have done:
- Kept the transition time
- Added the delay only if there is a real change in both brightness and color (so no effect on user experience when using the UIs)
- Added a boolean setting (by now set to true) so that this behaviour can be activated on demand once there is a way to provide conguration

Just testing, and will now send the PR with all the changes

Reply:
Or, considering XY support "experimental", I am setting the "delayedColorChange" boolean to true only for XY lights, so that we no not change the behaviour for working Hue/Sat lights. Would this make sense for you ?

Reply:
I don't mind either way. Is this issue only an XY issue, or also an HS issue? I assumed both?

At the end of the day, if there are "problems" after we merge this, then we'll sort them out. If it's not 100% perfect, then we'll learn from it and find the best way even if it takes a couple of tries and some wider user feedback...

Reply:
A transition time of 0 still requires some 100ms delay between the two commands, or the first one is acked, but silently aborted when receiving the second one. When there is a transition, I see the light start the level change, and then revert to the original brightness when it receives the color command. Nice bug...

I do not think the issue is only related to XY, but as there is surely people already using the channel for hue/sat lights, and nobody complained about it, to avoid impact on users for who it is working OK now, it may be better to apply this to the only XY light we have to test (for which it fails without the delay), and later on let this be configured if needed.